blog_post_collaboration/kedro_ibis.md

```diff
@@ -0,0 +1,263 @@
+
+> _"When I learned about Kedro (while at dbt Labs), I commented that it was like dbt if it were created by Python data scientists instead of SQL data analysts (including both being created out of consulting companies)."_
+>
+> —Cody Peterson, Technical Product Manager
+
+## "What if I don't want to use SQL?"
+
+This post isn't going to enter the SQL vs. Python debate. Nevertheless, many teams and data scientists choose Python for their analytics workflows. Furthermore, in my experience, teams using Kedro for their machine learning pipelines prefer to use the same tool for data processing and feature engineering code, too.
+
+Can we combine the flexibility and familiarity of Python with the scale and performance of modern SQL? Yes we can! ["Ibis is a Python library that provides a lightweight, universal interface for data wrangling."](https://github.com/ibis-project/ibis) Ibis exposes a Python dataframe API where the same code can execute against over 15 query engines, from local backends like pandas, Polars, and DuckDB to remote databases and distributed computation frameworks like BigQuery, Snowflake, and Spark. Ibis constructs a query plan, or intermediate representation (IR), that it evaluates lazily (i.e. as and when needed) on the execution engine.
+
+## Reimagining Python-first production data pipelines
+
+### Revisiting the Jaffle Shop
+
+Since dbt represents the industry standard for SQL-first transformation workflows, let's use the Jaffle Shop example to inform the capabilities of a Python-first solution. For those unfamiliar with the Jaffle Shop project, it provides a playground dbt project for testing and demonstration purposes, akin to the Kedro spaceflights project. Take a few minutes to [try dbt with DuckDB yourself](https://github.com/dbt-labs/jaffle_shop_duckdb) or [watch a brief walkthrough of the same](https://www.loom.com/share/ed4a6f59957e43158837eb4ba0c5ed67).
+
+If you want to peek at the final solution using Kedro at this point, [jump to the "Quickstart" section](#quickstart).
+
+### Creating a custom `ibis.Table` dataset
+
+For our Kedro-Ibis integration, we need the ability to load and save data using the backends Ibis supports. The data will be represented in memory as an [Ibis table](https://ibis-project.org/reference/expression-tables#ibis.expr.types.relations.Table), analagous to a pandas dataframe. The first step in using Ibis involves connecting to the desired backend; for example, see [how to connect to the DuckDB backend](https://ibis-project.org/backends/duckdb#connect). We abstract this step in the dataset configuration. We also maintain a mapping of established connections for reuse—a technique borrowed from existing dataset implementations, like [those of `pandas.SQLTableDataset` and `pandas.SQLQueryDataset`](https://github.com/kedro-org/kedro-plugins/blob/kedro-datasets-2.0.0/kedro-datasets/kedro_datasets/pandas/sql_dataset.py).
```

Comment:
s/analagous/analogous/g

Reply:
Also, AFAIK an Ibis table is lazy, rather than materialized? Hence it's not exactly "analogous" (worth picking a better word, given that the lazy nature was mentioned in the Ibis introduction paragraph)

Reply:
Call me lazy, but I just added "(lazy)". :D